src/token/TokenRejectTransaction.js

```diff
@@ -0,0 +1,270 @@
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+import AccountId from "../account/AccountId.js";
+import TokenId from "./TokenId.js";
+import NftId from "../token/NftId.js";
+import Transaction from "../transaction/Transaction.js";
+import { TRANSACTION_REGISTRY } from "../transaction/Transaction.js";
+
+/**
+ * @namespace proto
+ * @typedef {import("@hashgraph/proto").proto.ITransaction} HashgraphProto.proto.ITransaction
+ * @typedef {import("@hashgraph/proto").proto.ISignedTransaction} HashgraphProto.proto.ISignedTransaction
+ * @typedef {import("@hashgraph/proto").proto.ITransactionBody} HashgraphProto.proto.ITransactionBody
+ * @typedef {import("@hashgraph/proto").proto.ITransactionResponse} HashgraphProto.proto.ITransactionResponse
+ * @typedef {import("@hashgraph/proto").proto.TransactionBody} HashgraphProto.proto.TransactionBody
+ * @typedef {import("@hashgraph/proto").proto.ITokenRejectTransactionBody} HashgraphProto.proto.ITokenRejectTransactionBody
+ * @typedef {import("@hashgraph/proto").proto.TokenReference} HashgraphProto.proto.TokenReference
+ */
+
+/**
+ * @typedef {import("../channel/Channel.js").default} Channel
+ * @typedef {import("../client/Client.js").default<*, *>} Client
+ * @typedef {import("../transaction/TransactionId.js").default} TransactionId
+ */
+
+/**
+ * Reject a new Hedera™ crypto-currency token.
+ */
+export default class TokenRejectTransaction extends Transaction {
+    /**
+     *
+     * @param {object} [props]
+     * @param {AccountId | string | null} [props.owner]
+     * @param {HashgraphProto.proto.TokenReference[]} [props.rejections]
+     */
+    constructor(props = {}) {
+        super();
+
+        /**
+         * @private
+         * @type {?AccountId}
+         */
+        this._owner = null;
+
+        if (props.owner != null) {
+            this.setOwnerId(props.owner);
+        }
+
+        /**
+         * @private
+         * @type {TokenId[]}
+         */
+        this._tokenIds = [];
+
+        /**
+         * @private
+         * @type {NftId[]}
+         */
+        this._nftIds = [];
+
+        if (props.rejections != null) {
+            for (let rejection of props.rejections) {
+                if (rejection instanceof TokenId) {
+                    this._tokenIds.push(rejection);
+                } else if (rejection instanceof NftId) {
+                    this._nftIds.push(rejection);
+                }
+            }
+        }
```

Comment:
I think this logic should be in method `_fromProtobuf`. If you change the types in the constructor to be `tokenID` list and `nftID` list instead of `tokenReference` you can just assign the values here.